### DISCUSSION TITLE
pre-calculating and loading embeding vectors

### DISCUSSION THREAD

**Discussion by avafor** [MaartenGr/BERTopic#806] [Q&A]
Hi, 
I would like to use the BERTopic on the fly where the users could get topics in a reasonable amount of time for a large dataset, i.e. over 10k documents. In this case, it would make sense if I calculate the embedding vectors in advance, save them somewhere and when a new querry comes in, I only pass the embedding vectors to the BERTopic. Is such a functionality already implemented? 
Thank you very much.

**Comment by MaartenGr** [comment]
Yes, you can pre-calculate the embeddings for your documents and then supply them to BERTopic as follows:

```python
from bertopic import BERTopic

topic_model = BERTopic()
topics, probs = topic_model.fit_transform(docs, embeddings)
```

**Reply by avafor** [reply]
perfect! Thank you very much.
